.github/workflows/quality-control.yml

```
@@ -1,13 +1,78 @@
-name: Build
+name: Quality Control # should probably be renamed
 on:
-  push:
-    branches:
-      - main
   pull_request:
-    types: [opened, synchronize, reopened]
+    types: [ opened, synchronize, reopened ]
 jobs:
   build:
-    name: Build
+    name: CMake Build
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        compiler: [ clang-12, clang-13, clang-14, gcc-11 ]
+        build-type: [ Debug, Release ]
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+          fetch-depth: 0
+
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install ninja-build libsdl2-dev libcriterion-dev -y
+
+      - name: Build
+        run: |
+          cmake -G Ninja -DCMAKE_C_COMPILER=${{ matrix.compiler }} -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -DOPTIMIZE=3 .
+          ninja -j 2
```

Maybe using 
```suggestion
          ninja -j 0
```
could work, too. This will maximize the number of parallel jobs for Ninja

---

Nice, good catch!